src/jit-format/jit-format.cs

```diff
@@ -0,0 +1,682 @@
+                        }
+                        else if (option.StartsWith("/U"))
+                        {
+                            uOptions.Add(option.Replace("/U", "-U"));
+                        }
+                        else if (option.Contains("src\\" + dir) && option.EndsWith(".cpp"))
+                        {
+                            files.Add(option);
+                        }
+                    }
+
+                    using (StreamWriter compileCommandsWriter = File.AppendText(compileCommands))
+                    {
+                        foreach (string filename in files)
+                        {
+                            string file = filename.Replace("\\", "/");
+                            if (!first)
+                            {
+                                compileCommandsWriter.WriteLine(",");
+                            }
+
+                            compileCommandsWriter.WriteLine("    {");
+
+                            compileCommandsWriter.WriteLine("        \"directory\": \"" + iOptions[0].Replace("-I","") + "\",");
+                            compileCommandsWriter.Write("        \"command\": \"\\\"" + clExe + "\\\" ");
+
+
+                            compileCommandsWriter.Write("-target x86_64-pc-win32-msvc -fms-extensions -fms-compatibility -fmsc-version=1900 -fexceptions -fcxx-exceptions ");
+
+                            foreach (string iOption in iOptions)
+                            {
+                                compileCommandsWriter.Write(iOption + " ");
+                            }
+                            
+                            foreach (string dOption in dOptions)
+                            {
+                                compileCommandsWriter.Write(dOption + " ");
+                            }
+                            foreach (string uOption in uOptions)
+                            {
+                                compileCommandsWriter.Write(uOption + " ");
+                            }
+
+                            compileCommandsWriter.WriteLine(file + "\",");
+                            compileCommandsWriter.WriteLine("        \"file\": \"" + file + "\"");
+                            compileCommandsWriter.Write("    }");
+                            first = false;
+                        }
+                    }
+                }
+            }
+
+            using (StreamWriter compileCommandsWriter = File.AppendText(compileCommands))
+            {
+                compileCommandsWriter.WriteLine();
+                compileCommandsWriter.WriteLine("]");
+                compileCommandsWriter.WriteLine();
+            }
+        }
+
+        public static bool RunClangTidy(List<string> filenames, string compileCommands, bool fix, bool verbose)
+        {
+            bool formatOk = true;
+            string checks = "readability-braces*,modernize-use-nullptr";
+            string tidyFix = fix ? "-fix " : " ";
```

Comment:
Probably best to leave for a later change once we've decided what it will look like.